executor/aggfuncs/func_varpop.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type varPop4Float64 struct {
+	baseAggFunc
+}
+
+type partialResult4Float64 struct {
+	count    int64
+	sum      float64
+	variance float64
+}
+
+func (e *varPop4Float64) AllocPartialResult() PartialResult {
+	return PartialResult(&partialResult4Float64{})
+}
+
+func (e *varPop4Float64) ResetPartialResult(pr PartialResult) {
+	p := (*partialResult4Float64)(pr)
+	p.sum = 0
+	p.count = 0
+}
+
+func (e *varPop4Float64) AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error {
+	p := (*partialResult4Float64)(pr)
+	if p.count == 0 {
+		chk.AppendNull(e.ordinal)
+		return nil
+	}
+	varicance := p.variance / float64(p.count)
+	chk.AppendFloat64(e.ordinal, varicance)
+	return nil
+}
+
+func calculateIntermediate(count int64, sum float64, input float64, variance float64) float64 {
```

Comment:
Can we reuse the `calculateMerge` function?

Reply:
There is no reuse at present.

Reply:
Sorry, I clicked the wrong button "re-request" just now.

Reply:
We can call `calculateMerge (count, 1 , sum, input, variance, 0) `to get the same result?

Reply:
One denominator is “dstCountFloat64+srcCountFloat64”， but another is “count -1”.As you can see, if the srcCount is count and dstCount is 1, the denominator will be “1 + count”. Apparently, count +1 != count -1.

Reply:
You update `count` and `sum` before calling `calculateIntermediate`, but update them after calling `calculateMerge`, So it leads the denominator is different. 
IMHO, we can update `count` and `sum` after calling `calculateIntermediate` and change `count * (count - 1)` to `(count +1) * count` . The logic is same as the older, but now the two function denominator is same.

Reply:
It will get different result. Because in every step, the updated sum and count need to participate in calculating variance. If we update them after calling calculateIntermediate, it will get wrong result.